src/cleanup_rules/swift/rules.toml

```diff
@@ -513,6 +513,155 @@ replace_node = "ternary_block"
+    )
+    (#eq? @var "@hvariable")
+    (#not-eq? @val "@hvalue")
+)"""]
+
+# rule to delete assignments like a = true subject to the mentioned constraints
+[[rules]]
+name = "delete_parent_assignment"
+query = """(
+    (assignment
+        target: (directly_assignable_expression
+            [ (navigation_expression
+                target: (self_expression)
+                suffix: (navigation_suffix
+                    suffix: (simple_identifier) @avariable
+                )
+                )
+                (simple_identifier) @avariable
+            ]
+            )
+        result: (boolean_literal) @avalue
+    ) @assignment
+    (#eq? @avariable "@hvariable")
+)"""
+replace_node = "assignment"
+replace = ""
+holes = ["hvariable", "hvalue"]
+is_seed_rule = false
+
+# skip the rule if the variable is declared with a different value in the class scope
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = ["""(
+    (property_declaration
+        name: (pattern
+            bound_identifier: (simple_identifier) @var
+        )
+        value: (boolean_literal) @val
+    )@property_declaration
+    (#eq? @var "@avariable")
+    (#not-eq? @val "@avalue")
+)"""]
+
+# skip the rule if the variable is being assigned a different value in the class scope
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = ["""(
+    (assignment
+        target: (directly_assignable_expression
+            [   (navigation_expression
+                    target: (self_expression)
+                    suffix: (navigation_suffix
+                        suffix: (simple_identifier) @var
+                    )
+                )
+                (simple_identifier) @var
+            ]
+        )
+        result: (boolean_literal) @val
+    )
+    (#eq? @var "@avariable")
+    (#not-eq? @val "@avalue")
+)"""]
+
+# rule to replace the identifier with the value with the mentioned constraints
+[[rules]]
+name = "replace_identifier_with_value"
+query = """(
+    (simple_identifier) @identifier
+    (#eq? @identifier "@hvariable")
+)"""
+replace_node = "identifier"
+replace = "@hvalue"
+holes = ["hvariable", "hvalue"]
+is_seed_rule = false
+
+# skip the rule if the variable is one of the parameters of the function
+[[rules.constraints]]
+matcher = "(function_declaration) @fd"
+queries = ["""(
+    (parameter
+        name: (simple_identifier) @var
+    )
+    (#eq? @var "@identifier")
+)"""]
+
+# skip the rule if there is a declaration of the same variable
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = ["""(
+    (property_declaration
+        name: (pattern
+            bound_identifier: (simple_identifier) @variable
+        )
+        value: (boolean_literal) @value
+    )
+    (#eq? @variable "@identifier")
+)"""]
+
+# skip the rule if there is an assignment of the same variable
+[[rules.constraints]]
+matcher = "(class_declaration) @cd"
+queries = ["""(
+    (navigation_expression
```

Comment:
No sure if the comment and constraint match. 

Anyhoo, we can take also combine this with previous constraint (Since they use the same `matcher` like my recommendation for previous rule)